### DISCUSSION TITLE
form update fails to render when returning a 400 status code

### DISCUSSION THREAD

**Discussion by sraturi** [unpoly/unpoly#235] [Q&A]
Hello,
I am trying to use` up-submit` and `up-validation` in a form. `up-submit` and `up-validation` seem to work whenever  I am returning any status code other than 400/BAD_REQUEST.  For example, I can return any 200 code or even 4XX like 409 just not 400.
When I return the 400 status code, the page fails to render properly and I get a blob of text that is my HTML code.

```
<form action="/customer/new" method="POST" name="newCustomer" enctype="application/x-www-form-urlencoded"
          up-submit up-validate>
....
</form>
```

I am using unpoly 2.1 and the return response i send is of type [Response](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.html).

**Comment by adam12** [comment]
What happens if you don't' use Unpoly on this form (ie. remove `up-*` attributes from `<form>` tag)? Do you get the same raw HTML as text?

What is the content-type for this request in your browsers network console?

Is there anything in the Unpoly log? `up.log.enable()`

**Reply by sraturi** [reply]
Ah, I get the same raw HTML without the `up` attributes. The `content-type` is  `text/html;charset=UTF-8`

I also attached a screenshot of the logs.
![image](https://user-images.githubusercontent.com/25393450/125834596-b3433d2d-85b7-4425-b88b-8a2ed5ad419e.png)

**Reply by adam12** [reply]
Is the content being output escaped? ie. have you View Sourced it?

I don't think this is related to Unpoly.

**Reply by sraturi** [reply]
Ah, thanks. I think you are right, it's not Unpoly at this point. It seems to be related to the project framework itself.
